### PR TITLE
Feature/sg 850 support torch 2.0

### PR DESCRIPTION
**Pull request by shaydeci**
- Fixed some failing tests in build after moving it to run on 3.8.
- Moved build to run on 3.8 (minimum python version requirement for torch 2.0)
- Fixed missing arch_params.num classes that were missing for pose estimation regression test.
- Updated requirement, pip version in CI.

### CODE REVIEW

**Review by Louis-Dupont — APPROVED** [review]
LGTM

File: src/super_gradients/recipes/coco2017_pose_dekr_w32_no_dc.yaml [hunk 1]
```diff
@@ -55,6 +55,9 @@ ckpt_root_dir:
 train_dataloader: coco2017_pose_train
 val_dataloader: coco2017_pose_val
 
+arch_params:
```
Comment (ofrimasad) [inline src/super_gradients/recipes/coco2017_pose_dekr_w32_no_dc.yaml:58]: How is that related?

File: tests/unit_tests/detection_utils_test.py [hunk 2]
```diff
@@ -24,7 +24,7 @@ def setUp(self):
     def test_visualization(self):
 
         valid_loader = coco2017_val(dataloader_params={"batch_size": 16})
-        trainer = Trainer("visualization_test", device=self.device)
+        trainer = Trainer("visualization_test")
```
Comment (ofrimasad) [inline tests/unit_tests/detection_utils_test.py:27]: how is that related?

**Review by ofrimasad — COMMENTED** [review]
please see a few questions before merge

File: tests/unit_tests/detection_utils_test.py (hunk 2 above)
Comment (shaydeci) [reply tests/unit_tests/detection_utils_test.py:27]: Not related, but I ran all of our unit tests (including those outside the suite) and fixed anything that was crashing.
Passing a divece through trainer was deprecated and causes to crash since few versions ago.

File: src/super_gradients/recipes/coco2017_pose_dekr_w32_no_dc.yaml (hunk 1 above)
Comment (shaydeci) [reply src/super_gradients/recipes/coco2017_pose_dekr_w32_no_dc.yaml:58]: Not related.
I ran our regression tests, and this one was crashing because of missing arch_params.num_classes.
Thouht Itd be a good idea to fix this now and not forget about it later.
